runtime/src/accounts_db.rs

```diff
@@ -4520,18 +4511,13 @@ impl AccountsDb {
             }
             rewrite_elapsed.stop();
 
-            let keep_store_ids = if slot == current_ancient.slot() {
-                // this slot became an ancient append vec, so we want to keep the append vec storage for the slot
-                // and remove the old, normal append vec
-                vec![current_ancient.append_vec_id()]
-            } else {
+            if slot != current_ancient.slot() {
                 // all append vecs in this slot have been combined into an ancient append vec
                 dropped_roots.push(slot);
-                Vec::default()
-            };
+            }
 
             let (_remaining_stores, remove_old_stores_shrink) =
-                measure!(self.remove_old_stores_shrink(&keep_store_ids, &shrink_collect, slot));
+                measure!(self.remove_old_stores_shrink(&shrink_collect, slot));
```

Comment:
Clarifying the change for myself:

So we used to pass the append_vec_id of the current ancient (if anything). Now we pass the store_ids containing unique accounts we are combining, from shrink_collect.

Then our `should_retain` fn is if it's *not* in this set, i.e. we keep stores not in the shrink_collect results.

Reply:
I had the logic reversed on the ancient append vec side. We were passing the ids to keep, but we were retaining append vecs that WERE in the vec. This was opposite and wrong.